onnxruntime/core/graph/graph.cc

```diff
@@ -1060,7 +1138,7 @@ Status Graph::BuildConnections(std::unordered_set<std::string>& outer_scope_node
             // If this Graph was built manually, remove the implicit input from the graph outputs if it is present there
             // and not explicitly listed in the ordered graph outputs (as that implies we should leave it as an output).
             // If the Graph was loaded from a GraphProto, honor the explicit graph outputs and leave as is.
-            if (!loaded_from_model_file) {
+            if (!is_loaded_from_model_file_) { // TODO do we need to account for graph_outputs_manually_set_ here?
               graph_outputs_.erase(std::remove(graph_outputs_.begin(), graph_outputs_.end(), node_arg),
```

Comment:
No. SetGraphInputsOutputs is called by Resolve before BuildConnections so they've already been processed.